core/rfb.js

```diff
@@ -1108,13 +1108,8 @@ export default class RFB extends EventTargetMixin {
             this._sendMouse(x, y, this._mouseButtonMask);
         }
 
-        if (down) {
-            this._mouseButtonMask |= bmask;
-        } else {
-            this._mouseButtonMask &= ~bmask;
-        }
-
-        this._sendMouse(x, y, this._mouseButtonMask);
+        this._sendMouse(x, y, bmask);
+        this._mouseButtonMask = bmask;
```

Comment:
If you change the order of these two, then you can retain that `_sendMouse()` is always called with `this._mouseButtonMask`.